rotkehlchen/chain/ethereum/modules/pickle/pickle.py

```diff
@@ -0,0 +1,147 @@
+        ]
+        balance_calls = [
+            (PICKLE_DILL.address, dill_contract.encode(method_name='locked', arguments=[x]))
+            for x in addresses
+        ]
+        reward_outputs = multicall_2(
+            ethereum=self.ethereum,
+            require_success=False,
+            calls=rewards_calls,
+        )
+        dill_outputs = multicall_2(
+            ethereum=self.ethereum,
+            require_success=False,
+            calls=balance_calls,
+        )
+
+        pickle_price = Inquirer().find_usd_price(A_PICKLE)
+        for idx, output in enumerate(reward_outputs):
+            status_rewards, result = output
+            status_dill, result_dill = dill_outputs[idx]
+            address = addresses[idx]
+            if all((status_rewards, status_dill)):
+                try:
+                    rewards = rewards_contract.decode(result, 'claim', arguments=[address])
+                    dill_amounts = dill_contract.decode(
+                        result_dill,
+                        'locked',
+                        arguments=[address],
+                    )
+                    dill_rewards = token_normalized_value_decimals(
+                        token_amount=rewards[0],  # pylint: disable=unsubscriptable-object
+                        token_decimals=A_PICKLE.decimals,
+                    )
+                    dill_locked = token_normalized_value_decimals(
+                        token_amount=dill_amounts[0],  # pylint: disable=unsubscriptable-object
+                        token_decimals=A_PICKLE.decimals,
+                    )
+                    balance = DillBalance(
+                        dill_amount=AssetBalance(
+                            asset=A_PICKLE,
+                            balance=Balance(
+                                amount=dill_locked,
+                                usd_value=pickle_price * dill_locked,
+                            ),
+                        ),
+                        pending_rewards=AssetBalance(
+                            asset=A_PICKLE,
+                            balance=Balance(
+                                amount=dill_rewards,
+                                usd_value=pickle_price * dill_rewards,
+                            ),
+                        ),
+                        lock_time=deserialize_timestamp(dill_amounts[1]),  # noqa: E501 pylint: disable=unsubscriptable-object
+                    )
+                    api_output[address] = balance
+                except (DeserializationError, IndexError) as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to query dill information for address {address}. {str(e)}',
+                    )
+
+        return api_output
+
+    # -- Methods following the EthereumModule interface -- #
+    def on_startup(self) -> None:
+        pass
+
+    def on_account_addition(self, address: ChecksumEthAddress) -> Optional[List['AssetBalance']]:
```

Comment:
You should actually abstract the code of the addition of pickle into balances that you have here: https://github.com/rotki/rotki/pull/3384/files#diff-d84167c831e56f413a945eadae2d8d0300f14411c67824697a6ec1892eff6381R1415-R1418

and also add it here. This is a per-module callback that runs whenever a new account is added.